src/runtime/ranger/ranger_resource_policy.cpp

```diff
@@ -0,0 +1,70 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "ranger_resource_policy.h"
+
+namespace dsn {
+namespace ranger {
+
+bool policy_item::match(const access_type &ac_type, const std::string &user_name) const
+{
+    return (access_types & ac_type) != 0 && users.count(user_name) != 0;
```

Comment:
Is `ac_type` allowed to just include single type or multiple types ? Are following examples matched ?
```
access_types = READ | DROP, ac_type = READ | CREATE
access_types = WRITE, ac_type = READ | WRITE | CREATE
```

Reply:
One rpc request can only match one `ac_type`, so `access_types = READ | DROP` will not occur.